kong/router/expressions.lua

```diff
@@ -19,8 +24,35 @@ local PROTOCOLS_OVERRIDE = {
 }
 
 
-local function get_exp_and_priority(route)
+-- net.port => net.dst.port
+local function transform_expression(route)
   local exp = route.expression
+
+  if not exp then
+    return nil
+  end
+
+  if not exp:find("net.port", 1, true) then
+    return exp
+  end
+
+  -- there is "net.port" in expression
+
+  local new_exp = re_gsub(exp, NET_PORT_REG, NET_PORT_REPLACE, "jo")
+
+  if exp ~= new_exp then
+    ngx.log(ngx.WARN, "The field 'net.port' of expression is deprecated " ..
+                      "and will be removed in the upcoming major release, " ..
+                      "please use 'net.dst.port' instead.")
```

Comment:
The `net.port` is an undocumented feature, so it might be used by our other products, if the expression is not written by the customer directly (such as KIC), this message might confuse them.

We should consider the message carefully.